istio/discovery.go

```diff
@@ -292,6 +295,38 @@ func NewDiscovery(clients map[string]kubernetes.ClientInterface, cache cache.Kia
 	}
 }
 
+// GetControlPlaneNamespaces returns control plane namespaces for the cluster. If cluster == "" then
+// it is for all clusters. it returns an empty slice.
+func (in *Discovery) GetControlPlaneNamespaces(ctx context.Context, cluster string) []string {
+	namespaces := map[string]bool{}
+
+	mesh, err := in.Mesh(ctx)
+	if err != nil {
+		log.Debugf("Unable to get mesh to determine control plane namespaces for cluster [%s]. Err: %s", cluster, err)
+		return maps.Keys(namespaces)
+	}
+
+	for _, cp := range mesh.ControlPlanes {
+		if cluster == "" || cluster == cp.Cluster.Name {
+			namespaces[cp.IstiodNamespace] = true
+		}
+	}
+
+	return maps.Keys(namespaces)
+}
+
+// IsControlPlane returns true if the cluster-namespace is an istio control plane. If cluster == "" it
+// is ignored, and only the namespace is considered. Otherwise false.
+func (in *Discovery) IsControlPlane(ctx context.Context, cluster, namespace string) bool {
+	for _, cpns := range in.GetControlPlaneNamespaces(ctx, cluster) {
+		if namespace == cpns {
+			return true
+		}
+	}
+
+	return false
+}
```

Comment:
Unit tests?

Reply:
Hmm, yes. I was thinking about the new CI workflow but should add some unit tests for some of these new services.